### ISSUE TITLE
high memory consuming during the training

### ISSUE THREAD

**Issue by chenkehai** [OpenNMT/OpenNMT-py#851]
When I use the OpenNMT-py to train a model, there is aways high memory consuming (20%, about 20GB).
How to solve this problem？

**Comment by vince62s** [comment]
look into other issues regarding sharding.

**Comment by chenkehai** [comment]
hi, vince62s,
Thanks a lot.
Could you give me an effective solution?

**Comment by rbommasanimozilla** [comment]
what size shards are you using? The recommended size is 131mb (corresponding to ~1M segments), if this is too big, try smaller shards

**Comment by chenkehai** [comment]
Hi, rbommasanimozilla,

I should use the default shards size. My training command line follows:

CUDA_VISIBLE_DEVICES=0 python train.py -data ./data/corpus.train -save_model ./model_4096/model -gpuid 0 \
        -layers 6 -rnn_size 512 -word_vec_size 512   \
        -encoder_type transformer -decoder_type transformer -position_encoding \
        -train_steps 100000 -max_generator_batches 16 -dropout 0.1 \
        -batch_size 4096 -batch_type tokens -normalization tokens -accum_count 4 \
        -optim adam -adam_beta2 0.998 -decay_method noam -warmup_steps 8000 -learning_rate 2 \
        -max_grad_norm 0 -param_init 0 -param_init_glorot \
        -label_smoothing 0.1 -report_every 100 \
        -save_checkpoint_steps 2000 -valid_steps 2000
The terminal follows:
![20180726155644](https://user-images.githubusercontent.com/5715046/43249159-8fd5fcd6-90ec-11e8-9df7-7fd27b22e9de.png)

**Comment by chenkehai** [comment]
Thank you very much!
